lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1609,7 +1609,6 @@ void CodeGenModule::SetFunctionAttributes(GlobalDecl GD, llvm::Function *F,
   if (getLangOpts().OpenCL ||
       (getLangOpts().CPlusPlusAMP && CodeGenOpts.AMPIsDevice)) {
       if (F->getName()=="amp_barrier") {
-          F->addFnAttr(llvm::Attribute::NoDuplicate);
           F->addFnAttr(llvm::Attribute::NoUnwind);
```

Comment:
NoUnwind is probably not needed here as well so let's remove the specialization for `amp_barrier` altogther